x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/CacheFile.java

```diff
@@ -305,6 +312,7 @@ protected void doRun() throws Exception {
                             reference.decRef();
                         }
                         gap.onCompletion();
+                        fsynced.set(false);
```

Comment:
Might be better to set this `false` as early as possible (i.e. directly after the write to the file) to reduce contention statistically speaking? (since `fsync` and `write` will always be blocking each other anyway even if we do use multiple `FileChannel` instances I think).

Reply:
I like that `fsynced` not only indicate that something was written to the file but that ranges are also updated - although it is not a requirement. I'm not sure if it makes a big difference but I can move this earlier.

Reply:
Ah that's a good point :)